lib/i18n/backend/cache.rb

```diff
@@ -60,12 +66,26 @@ def perform_caching?
   end
 
   module Backend
-    # TODO Should the cache be cleared if new translations are stored?
     module Cache
+      @@cached_keys = []
+
+      def store_translations(locale, data, options = {})
+        clear_cache if I18n.perform_caching?
```

Comment:
Running `I18n.cache_store.cleanup` will be enough. Everything else should be done in a cache store class: http://api.rubyonrails.org/classes/ActiveSupport/Cache/Store.html#method-i-cleanup